Tests/PostgresNIOTests/New/Messages/DataRowTests.swift

```diff
@@ -20,18 +20,127 @@ class DataRowTests: XCTestCase {
             decoderFactory: { PSQLBackendMessageDecoder(hasAlreadyReceivedBytes: false) }))
     }
+    
+    func testIteratingElements() {
+        let dataRow = DataRow.makeTestDataRow(nil, ByteBuffer(), ByteBuffer(repeating: 5, count: 10))
+        var iterator = dataRow.makeIterator()
+        
+        XCTAssertEqual(dataRow.count, 3)
+        XCTAssertEqual(iterator.next(), .some(.none))
+        XCTAssertEqual(iterator.next(), ByteBuffer())
+        XCTAssertEqual(iterator.next(), ByteBuffer(repeating: 5, count: 10))
+        XCTAssertEqual(iterator.next(), .none)
+    }
+    
+    func testIndexAfterAndSubscript() {
+        let dataRow = DataRow.makeTestDataRow(
+            nil,
+            ByteBuffer(),
+            ByteBuffer(repeating: 5, count: 10),
+            nil
+        )
+        
+        var index = dataRow.startIndex
+        XCTAssertEqual(dataRow[index], .none)
+        index = dataRow.index(after: index)
+        XCTAssertEqual(dataRow[index], ByteBuffer())
+        index = dataRow.index(after: index)
+        XCTAssertEqual(dataRow[index], ByteBuffer(repeating: 5, count: 10))
+        index = dataRow.index(after: index)
+        XCTAssertEqual(dataRow[index], .none)
+        index = dataRow.index(after: index)
+        XCTAssertEqual(index, dataRow.endIndex)
+    }
+    
+    func testIndexComparison() {
+        let dataRow = DataRow.makeTestDataRow(
+            nil,
+            ByteBuffer(),
+            ByteBuffer(repeating: 5, count: 10),
+            nil
+        )
+        
+        let startIndex = dataRow.startIndex
+        let secondIndex = dataRow.index(after: startIndex)
+        
+        XCTAssertLessThanOrEqual(startIndex, secondIndex)
+        XCTAssertLessThan(startIndex, secondIndex)
+        
+        XCTAssertGreaterThanOrEqual(secondIndex, startIndex)
+        XCTAssertGreaterThan(secondIndex, startIndex)
+        
+        XCTAssertFalse(secondIndex == startIndex)
+        XCTAssertEqual(secondIndex, secondIndex)
+        XCTAssertEqual(startIndex, startIndex)
+    }
+    
+    func testColumnSubscript() {
+        let dataRow = DataRow.makeTestDataRow(
+            nil,
+            ByteBuffer(),
+            ByteBuffer(repeating: 5, count: 10),
+            nil
+        )
+    
+        XCTAssertEqual(dataRow.count, 4)
+        XCTAssertEqual(dataRow[column: 0], .none)
+        XCTAssertEqual(dataRow[column: 1], ByteBuffer())
+        XCTAssertEqual(dataRow[column: 2], ByteBuffer(repeating: 5, count: 10))
+        XCTAssertEqual(dataRow[column: 3], .none)
+    }
+    
+    func testWithContiguousStorageIfAvailable() {
+        let dataRow = DataRow.makeTestDataRow(
+            nil,
+            ByteBuffer(),
+            ByteBuffer(repeating: 5, count: 10),
+            nil
+        )
+        
+        XCTAssertNil(dataRow.withContiguousStorageIfAvailable { _ -> Int in
+            XCTFail("DataRow does not have a contiguous storage")
+            return 123
+        })
```

Comment:
```suggestion
        XCTAssertNil(dataRow.withContiguousStorageIfAvailable { _ in
            return XCTFail("DataRow should not be able to offer contiguous storage")
        })
```

Reply:
The nil check only really works, if we return a value from the closure. If we return Void, the nil check doesn't do anything.